nativelib/src/main/scala/scala/scalanative/native/complex.scala

```diff
@@ -0,0 +1,209 @@
+package scala.scalanative
+package native
+
+/** All methods take complex but Scala Native does not
+ * support pass by value so we pass a pointer to a
+ * Array of length 2 and have a small wrapper in C
+ * to do the conversion. Currently Scala Native
+ * and JVM have no direct support for long double
+ * so these methods are not implemented.
+ *
+ * Warning: Ptr[CComplexFloat] and Ptr[CComplex] values
+ * passed in to the functions are mutated for
+ * memory safety. Make copies on the stack or heap
+ * for values you need after the functions are called.
+ * Implicit classes are provided for convenience.
+ *
+ *
+ * References:
+ * https://en.wikipedia.org/wiki/C_data_types
+ * C99 also added complex types: float _Complex, double _Complex, long double _Complex
+ * https://en.wikipedia.org/wiki/Long_double
+ * http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/complex.h.html
+ *
+ */
+@extern
+object complex {
+  import Nat._2
+  type CComplexFloat = CArray[CFloat, _2]
+  type CComplex      = CArray[CDouble, _2]
+
+  @name("scalanative_cacosf")
+  def cacosf(complex: Ptr[CComplexFloat]): Ptr[CComplexFloat] = extern
```

Comment:
It seems like you use the same memory as the argument and overwrite it with the result. Given this semantics I'd return `Unit` instead, otherwise `Ptr[...]` as a return type is extremely deceiving: it's a not a pure mathematical function!